packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/arrow.dart

```diff
@@ -0,0 +1,280 @@
+        super(key: key);
+
+  final Color arrowColor;
+  final double arrowHeadSize;
+  final double arrowStrokeWidth;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: <Widget>[
+        Expanded(
+          child: ArrowWidget(
+            color: arrowColor,
+            headSize: arrowHeadSize,
+            strokeWidth: arrowStrokeWidth,
+            type: ArrowType.left,
+          ),
+        ),
+        Container(
+          child: child,
+          margin: const EdgeInsets.symmetric(horizontal: 8.0),
+        ),
+        Expanded(
+          child: ArrowWidget(
+            color: arrowColor,
+            headSize: arrowHeadSize,
+            strokeWidth: arrowStrokeWidth,
+            type: ArrowType.right,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// Widget that wraps and center another widget vertically with bidirectional arrow (up & down)
+@immutable
+class BidirectionalVerticalArrowWrapper extends StatelessWidget {
+  const BidirectionalVerticalArrowWrapper({
+    Key key,
+    @required this.child,
+    this.arrowColor = Colors.white,
+    this.arrowHeadSize = 16.0,
+    this.arrowStrokeWidth = 2.0,
+  })  : assert(child != null),
+        super(key: key);
+
+  final Color arrowColor;
+  final double arrowHeadSize;
+  final double arrowStrokeWidth;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: <Widget>[
+        Expanded(
+          child: ArrowWidget(
+            color: arrowColor,
+            headSize: arrowHeadSize,
+            strokeWidth: arrowStrokeWidth,
+            type: ArrowType.up,
+          ),
+        ),
+        Container(
+          child: child,
+          margin: const EdgeInsets.symmetric(vertical: 8.0),
+        ),
+        Expanded(
+          child: ArrowWidget(
+            color: arrowColor,
+            headSize: arrowHeadSize,
+            strokeWidth: arrowStrokeWidth,
+            type: ArrowType.down,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// Widget that draws a fully sized, centered, unidirectional arrow according to its constraints
+@immutable
+class ArrowWidget extends StatelessWidget {
+  const ArrowWidget({
+    this.color = Colors.white,
+    this.headSize = 16.0,
+    Key key,
+    this.strokeWidth = 2.0,
+    @required this.type,
+  })  : assert(color != null),
+        assert(headSize != null),
+        assert(strokeWidth != null),
+        assert(type != null),
+        super(key: key);
+
+  final Color color;
+
+  /// The arrow head is a Equilateral triangle
+  final double headSize;
+
+  final double strokeWidth;
+
+  final ArrowType type;
+
+  CustomPainter get _painter => _ArrowPainter(
```

Comment:
nit: make this a final field instead of a getter as this is a stateless widget so it can't actually change anyway.